### PR TITLE
Change storage file to track2

### PR DESCRIPTION
**Pull request by Luyunmt**
SDK Updated List

Azure.Storage.Blobs
Azure.Storage.Common
Azure.Storage.Files.Shares
@jongio for notification.

### CODE REVIEW

**Comment by v-xuto** [comment]
@jongio - Can you help ping the corresponding person to review this PR?

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @kyle-patterson , @tg-msft, @amishra-dev - Could you help review this PR?

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @kyle-patterson , @tg-msft, @amishra-dev - Could you help review this PR?

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @kyle-patterson , @tg-msft, @amishra-dev - Could you help review this PR?

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @kyle-patterson , @tg-msft, @amishra-dev - Could you help review this PR?

File: FileStorage/GettingStarted.cs [hunk 1]
```diff
@@ -152,27 +161,27 @@ public async Task RunFileStorageOperationsAsync()
...
+                fileList = fileDirectory.GetFilesAndDirectoriesAsync();
 
                 // Print all files/directories in the folder.
-                foreach (IListFileItem listItem in fileList)
+                await foreach (ShareFileItem listItem in fileList)
                 {
-                    // listItem type will be CloudFile or CloudFileDirectory.
-                    Console.WriteLine("    - {0} (type: {1})", listItem.Uri, listItem.GetType());
+                    // listItem type will be ShareClient or ShareDirectoryClient.
+                    Console.WriteLine("    - {0} (type: {1})", listItem.Name, listItem.GetType());
```
Comment (christothes) [inline FileStorage/GettingStarted.cs]: I think we should use IsDirectory here instead

```suggestion
                    // listItem type will be a file or directory
                    Console.WriteLine("    - {0} (IsDirectory: {1})", listItem.Name, listItem.IsDirectory);
```

File: FileStorage/GettingStarted.cs [hunk 2]
```diff
@@ -198,87 +214,105 @@ public async Task RunFileStorageOperationsAsync()
...
+                                    await targetBlob.AbortCopyFromUriAsync(copyId);
+                                    Console.WriteLine("   Cancelling the copy succeeded.");
+                                }
+                                else
+                                {
+                                    // If this happens, try a larger file.
+                                    Console.WriteLine("    Cancellation of copy not performed; copy has already finished.");
+                                }
+                
+                */
```
Comment (christothes) [inline FileStorage/GettingStarted.cs]: Is this intentionally commented out?

**Comment by Tianyun-G** [comment]
@christothes - I have updated ,please review.

File: FileStorage/Advanced.cs [hunk 3]
```diff
@@ -37,48 +35,47 @@ public async Task RunFileStorageAdvancedOpsAsync()
...
             // How to create a storage connection string - http://msdn.microsoft.com/en-us/library/azure/ee758697.aspx
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(CloudConfigurationManager.GetSetting("StorageConnectionString"));
+            string storageConnectionString = ConfigurationManager.AppSettings.Get("StorageConnectionString");
 
             Console.WriteLine("Instantiating file client.");
             Console.WriteLine(string.Empty);
 
-            // Create a file client for interacting with the file service.
-            CloudFileClient cloudFileClient = storageAccount.CreateCloudFileClient();
+            // Create a share client for interacting with the file service.
```
Comment (christothes) [inline FileStorage/Advanced.cs]:
```suggestion
            // Create a share service client for interacting with the file service.
```

File: FileStorage/packages.config [hunk 4]
```diff
@@ -1,13 +1,24 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.KeyVault.Core" version="3.0.3" targetFramework="net452" />
-  <package id="Microsoft.Azure.Storage.Common" version="10.0.2" targetFramework="net472" />
-  <package id="Microsoft.Azure.Storage.File" version="10.0.2" targetFramework="net472" />
+  <package id="Azure.Core" version="1.18.0" targetFramework="net472" />
```
Comment (christothes) [inline FileStorage/packages.config]: remove?

**Comment by v-xuto** [comment]
@christothes - We have updated by your comments, please rereview this PR.

**Comment by christothes** [comment]
Please update to the latest released versions of the storage packages and clean up any unneeded dependencies from the project and packages.config.

**Comment by Tianyun-G** [comment]
@christothes - I have updated , please review.

**Comment by v-xuto** [comment]
@christothes - We have updated by your comments, please re-review this PR.

**Review by christothes — APPROVED** [review]

File: FileStorage/packages.config (hunk 4 above)
Comment (Tianyun-G) [reply FileStorage/packages.config]: @christothes - I have removed, please review.

**Review by christothes — APPROVED** [review]

File: FileStorage/Advanced.cs [hunk 5]
```diff
@@ -37,48 +35,47 @@ public async Task RunFileStorageAdvancedOpsAsync()
...
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(CloudConfigurationManager.GetSetting("StorageConnectionString"));
+            string storageConnectionString = ConfigurationManager.AppSettings.Get("StorageConnectionString");
 
             Console.WriteLine("Instantiating file client.");
             Console.WriteLine(string.Empty);
 
-            // Create a file client for interacting with the file service.
-            CloudFileClient cloudFileClient = storageAccount.CreateCloudFileClient();
+            // Create a share service client for interacting with the file service.
+            ShareServiceClient shareServiceClient = new ShareServiceClient(storageConnectionString);
```
Comment (scottaddie) [inline FileStorage/Advanced.cs]: Prefer `var` when the type is obvious.

```suggestion
            var shareServiceClient = new ShareServiceClient(storageConnectionString);
```

File: FileStorage/Advanced.cs [hunk 6]
```diff
@@ -146,91 +142,96 @@ private static async Task ListSharesSample(CloudFileClient cloudFileClient)
...
             Console.WriteLine("Get service properties");
-            FileServiceProperties originalProperties = await fileClient.GetServicePropertiesAsync();
+            ShareServiceProperties originalProperties = await shareServiceClient.GetPropertiesAsync();
             try
             {
                 // Add CORS rule
                 Console.WriteLine("Add CORS rule");
 
-                CorsRule corsRule = new CorsRule
+                ShareCorsRule corsRule = new ShareCorsRule
```
Comment (scottaddie) [inline FileStorage/Advanced.cs]:
```suggestion
                var corsRule = new ShareCorsRule
```

File: FileStorage/Advanced.cs [hunk 7]
```diff
@@ -358,58 +364,59 @@ private static async Task DirectoryPropertiesSample(CloudFileClient cloudFileCli
...
                 // Set directory metadata
                 Console.WriteLine("Set directory metadata");
-                sampleDirectory.Metadata.Add("key1", "value1");
-                sampleDirectory.Metadata.Add("key2", "value2");
-                await sampleDirectory.SetMetadataAsync();
+
+                Dictionary<string, string> metadate = new Dictionary<string, string>();
+                metadate.Add("key1", "value1");
+                metadate.Add("key2", "value2");
+                await sampleDirectory.SetMetadataAsync(metadate);
```
Comment (scottaddie) [inline FileStorage/Advanced.cs]:
```suggestion
                var metadata = new Dictionary<string, string> { {"key1", "value1"}, {"key2", "value2"} };
                await sampleDirectory.SetMetadataAsync(metadata);
```

File: FileStorage/Advanced.cs [hunk 8]
```diff
@@ -426,59 +433,59 @@ private static async Task DirectoryMetadataSample(CloudFileClient cloudFileClien
...
+                
+                ShareFileClient file = rootDirectory.GetFileClient("demofile");
 
                 // Create file
                 Console.WriteLine("Create file");
                 await file.CreateAsync(1000);
-                
+
+                // Set file properties
+                ShareFileHttpHeaders headers = new ShareFileHttpHeaders()
```
Comment (scottaddie) [inline FileStorage/Advanced.cs]:
```suggestion
                var headers = new ShareFileHttpHeaders
```

File: FileStorage/Advanced.cs [hunk 9]
```diff
@@ -495,57 +502,60 @@ private static async Task FilePropertiesSample(CloudFileClient cloudFileClient)
...
+                Console.WriteLine("Create file");
+                await file.CreateAsync(1000);
 
                 // Set file metadata
                 Console.WriteLine("Set file metadata");
-                file.Metadata.Add("key1", "value1");
-                file.Metadata.Add("key2", "value2");
+                Dictionary<string, string> metadata = new Dictionary<string, string>();
+                metadata.Add("key1", "value1");
+                metadata.Add("key2", "value2");
```
Comment (scottaddie) [inline FileStorage/Advanced.cs]:
```suggestion
                var metadata = new Dictionary<string, string> { {"key1", "value1"}, {"key2", "value2"} };
```

File: FileStorage/Advanced.cs [hunk 10]
```diff
@@ -247,49 +248,55 @@ private static async Task SharePropertiesSample(CloudFileClient cloudFileClient)
...
 
             // Set share metadata
             Console.WriteLine("Set share metadata");
-            share.Metadata.Add("key1", "value1");
-            share.Metadata.Add("key2", "value2");
 
+            Dictionary<string, string> metaData = new Dictionary<string, string>();
+            metaData.Add("key1", "key1");
+            metaData.Add("key2", "key2");
+            await shareClient.SetMetadataAsync(metaData);
```
Comment (scottaddie) [inline FileStorage/Advanced.cs]:
```suggestion
            var metadata = new Dictionary<string, string> { {"key1", "value1"}, {"key2", "value2"} };
            await shareClient.SetMetadataAsync(metadata);
```

**Review by scottaddie — CHANGES_REQUESTED** [review]

**Comment by Tianyun-G** [comment]
@scottaddie - I have updated , please review.

File: FileStorage/Advanced.cs [hunk 11]
```diff
@@ -123,12 +119,12 @@ private static async Task ListSharesSample(CloudFileClient cloudFileClient)
                 Console.WriteLine("List of shares in the storage account:");
 
                 // List the file shares for this storage account 
-                IEnumerable<CloudFileShare> cloudShareList = cloudFileClient.ListShares();
+                var shareList = shareServiceClient.GetSharesAsync();
                 try
                 {
-                    foreach (CloudFileShare cloudShare in cloudShareList)
+                    await foreach (ShareItem Share in shareList)
```
Comment (scottaddie) [inline FileStorage/Advanced.cs]:
```suggestion
                    await foreach (ShareItem share in shareList)
```

File: FileStorage/Advanced.cs [hunk 12]
```diff
@@ -123,12 +119,12 @@ private static async Task ListSharesSample(CloudFileClient cloudFileClient)
...
                     {
-                        Console.WriteLine("Cloud Share name = {0}", cloudShare.Name);
+                        Console.WriteLine("Cloud Share name = {0}", Share.Name);
```
Comment (scottaddie) [inline FileStorage/Advanced.cs]:
```suggestion
                        Console.WriteLine("Cloud Share name = {0}", share.Name);
```

File: FileStorage/GettingStarted.cs [hunk 13]
```diff
@@ -35,28 +38,31 @@ public class GettingStarted
...
+                string storageConnectionString = ConfigurationManager.AppSettings.Get("StorageConnectionString");
+                string storageAccountName = ConfigurationManager.AppSettings.Get("StorageAccountName");
+                string storageAccountKey = ConfigurationManager.AppSettings.Get("StorageAccountKey");
 
                 Console.WriteLine("Instantiating file client.");
 
-                // Create a file client for interacting with the file service.
-                CloudFileClient cloudFileClient = storageAccount.CreateCloudFileClient();
+                // Create a share client for interacting with the file service.
+                ShareServiceClient shareServiceClient = new ShareServiceClient(storageConnectionString);
```
Comment (scottaddie) [inline FileStorage/GettingStarted.cs]:
```suggestion
                var shareServiceClient = new ShareServiceClient(storageConnectionString);
```

File: FileStorage/GettingStarted.cs [hunk 14]
```diff
@@ -198,87 +213,78 @@ public async Task RunFileStorageOperationsAsync()
...
+                }
                 Console.WriteLine("    Successfully uploaded file to share.");
 
                 // Copy the file to blob storage.
                 Console.WriteLine("Copying file to blob storage. Container name = {0}", shareName);
 
-                // First get a reference to the blob. 
-                CloudBlobClient cloudBlobClient = storageAccount.CreateCloudBlobClient();
+                // First get a blob service client. 
+                BlobServiceClient blobServiceClient = new BlobServiceClient(storageConnectionString);
```
Comment (scottaddie) [inline FileStorage/GettingStarted.cs]:
```suggestion
                var blobServiceClient = new BlobServiceClient(storageConnectionString);
```

File: FileStorage/GettingStarted.cs [hunk 15]
```diff
@@ -198,87 +213,78 @@ public async Task RunFileStorageOperationsAsync()
...
-                cloudFile = fileDirectory.GetFileReference(testFile);
+                // Get a share file client to be copied.
+                shareFileClient = fileDirectory.GetFileClient(testFile);
 
                 // Create a SAS for the file that's valid for 24 hours.
                 // Note that when you are copying a file to a blob, or a blob to a file, you must use a SAS
                 // to authenticate access to the source object, even if you are copying within the same
                 // storage account.
-                string fileSas = cloudFile.GetSharedAccessSignature(new SharedAccessFilePolicy()
+                AccountSasBuilder sas = new AccountSasBuilder
```
Comment (scottaddie) [inline FileStorage/GettingStarted.cs]:
```suggestion
                var sas = new AccountSasBuilder
```

File: FileStorage/GettingStarted.cs [hunk 16]
```diff
@@ -198,87 +213,78 @@ public async Task RunFileStorageOperationsAsync()
...
+                    // Access expires in 1 day!
+                    ExpiresOn = DateTime.UtcNow.AddDays(1)
+                };
+                sas.SetPermissions(AccountSasPermissions.Read);
 
-                // Print out the copy state information.
-                targetBlob.FetchAttributes();
-                Console.WriteLine("    targetBlob.copystate.CopyId = {0}", targetBlob.CopyState.CopyId);
-                Console.WriteLine("    targetBlob.copystate.Status = {0}", targetBlob.CopyState.Status);
+                StorageSharedKeyCredential credential = new StorageSharedKeyCredential(storageAccountName, storageAccountKey);
```
Comment (scottaddie) [inline FileStorage/GettingStarted.cs]:
```suggestion
                var credential = new StorageSharedKeyCredential(storageAccountName, storageAccountKey);
```

File: FileStorage/GettingStarted.cs [hunk 17]
```diff
@@ -198,87 +213,78 @@ public async Task RunFileStorageOperationsAsync()
...
-                {
-                    Console.WriteLine("   Status is Pending; cancelling the copy operation.");
-                    // Call to stop the copy, passing in the copyId of the operation.
-                    // This won't work if it has already finished copying.
-                    await targetBlob.AbortCopyAsync(copyId);
-                    Console.WriteLine("   Cancelling the copy succeeded.");
-                }
-                else
+                // Build a SAS URI
+                UriBuilder sasUri = new UriBuilder(shareFileClient.Uri)
```
Comment (scottaddie) [inline FileStorage/GettingStarted.cs]:
```suggestion
                var sasUri = new UriBuilder(shareFileClient.Uri)
```

File: FileStorage/GettingStarted.cs [hunk 18]
```diff
@@ -319,23 +322,29 @@ public async Task RunFileStorageOperationsAsync()
...
                         // Create a file with a maximum file size of 64k. 
-                        await cloudFile.CreateAsync(maxFileSize);
+                        await shareFileClient.CreateAsync(maxFileSize);
                         Console.WriteLine("    Empty file created successfully.");
                     }
 
                     // Write the stream to the file starting at startOffset for the length of the stream.
                     Console.WriteLine("Writing range to file.");
-                    await cloudFile.WriteRangeAsync(ms, startOffset, null);
+                    HttpRange range = new HttpRange(startOffset, textToStream.Length);
```
Comment (scottaddie) [inline FileStorage/GettingStarted.cs]:
```suggestion
                    var range = new HttpRange(startOffset, textToStream.Length);
```

File: FileStorage/GettingStarted.cs [hunk 19]
```diff
@@ -362,28 +371,36 @@ public async Task RunFileStorageOperationsAsync()
 
                     // Write the stream to the file starting at startOffset for the length of the stream.
                     Console.WriteLine("Write second range to file.");
-                    await cloudFile.WriteRangeAsync(ms, startOffset, null);
+                    HttpRange range = new HttpRange(startOffset, textToStream.Length);
```
Comment (scottaddie) [inline FileStorage/GettingStarted.cs]:
```suggestion
                    var range = new HttpRange(startOffset, textToStream.Length);
```

**Review by scottaddie — APPROVED** [review]
This PR looks good after incorporating these suggestions.

**Comment by Tianyun-G** [comment]
@scottaddie - I have updated , please review.

**Review by scottaddie — APPROVED** [review]

**Comment by v-xuto** [comment]
@jongio Please help to merge this PR.

**Pull request merged**